bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVaneerSettingsFactory.java

```diff
@@ -0,0 +1,302 @@
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.FixedTransportChannelProvider;
+import com.google.auth.Credentials;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminSettings;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings.Builder;
+import com.google.cloud.bigtable.data.v2.models.InstanceName;
+import com.google.cloud.bigtable.data.v2.stub.BigtableStubSettings;
+import com.google.cloud.bigtable.grpc.BigtableSession;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import org.threeten.bp.Duration;
+
+import static com.google.api.client.util.Preconditions.checkState;
+import static com.google.cloud.bigtable.config.CredentialOptions.getEnvJsonFile;
+import static org.threeten.bp.Duration.ofMillis;
+
+/**
+ * Static methods to convert an instance of {@link BigtableOptions} to a
+ * {@link BigtableDataSettings} or {@link BigtableTableAdminSettings} instance .
+ */
+public class BigtableVaneerSettingsFactory {
+  /** Constant <code>LOG</code> */
+  private static final Logger LOG = new Logger(BigtableVaneerSettingsFactory.class);
+
+  /**
+   * To create an instance of {@link BigtableDataSettings} from {@link BigtableOptions}.
+   *
+   * @param options a {@link BigtableOptions} object.
+   * @return a {@link BigtableDataSettings} object.
+   * @throws IOException if any.
+   */
+  public static BigtableDataSettings fromBigtableOptions(final BigtableOptions options) throws IOException {
+    checkState(options.getProjectId() != null, "Project ID is required");
+    checkState(options.getInstanceId() != null, "Instance ID is required");
+    checkState(options.getRetryOptions().enableRetries(), "Disabling retries is not currently supported.");
+
+    BigtableDataSettings.Builder builder = BigtableDataSettings.newBuilder();
+
+    InstanceName instanceName = InstanceName.newBuilder().setProject(options.getProjectId())
+        .setInstance(options.getInstanceId()).build();
+    builder.setInstanceName(instanceName);
+    builder.setAppProfileId(options.getAppProfileId());
+
+    builder.setEndpoint(options.getDataHost() + ":" + options.getPort());
+
+    builder.setCredentialsProvider(buildCredentialProvider(options.getCredentialOptions()));
+
+    buildBulkMutationsSettings(builder, options);
+
+    buildCheckAndMutateRowSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadModifyWriteSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadRowsSettings(builder, options);
+
+    buildMutateRowSettings(builder, options);
+
+    buildSampleRowKeysSettings(builder, options);
+
+    ManagedChannel channel = null;
+    try{
+      channel = BigtableSession.createChannelPool(options.getDataHost(), options);
```

Comment:
Utilizing BigtableSession#createChannelPool so that channelPoolCount is provided in the created channel.